pyreisejl/utility/state.py

```diff
@@ -1,24 +1,68 @@
 from dataclasses import dataclass, field
-from typing import Any, Dict
+from queue import Empty, Queue
+from threading import Thread
+from typing import Any, Dict, List
 
 from pyreisejl.utility.helpers import get_scenario_status
 
 
+class Listener:
+    """Runs in the background to read from stdout/stderr of a long lived
+    process"""
+
+    def __init__(self, stream):
+        self.stream = stream
+        self.queue = Queue()
```

Comment:
Unbounded Queue leaves you vulnerable to OOM issues. Would it be better to limit and throw an exception when out of bounds, or shed oldest entries?

Reply:
Good point, not sure how much of an issue it is but I will make a note to revisit this. For context, the size of the streams is much smaller than the output data, which is most likely to cause OOM issues.